tests/unit/unit-util.cpp

```diff
@@ -156,6 +156,64 @@ TEST_P(UnitUtilIntTests, Oracles)
   }
 }
 
+TEST_P(UnitUtilTests, cnf_to_dimacs)
+{
+  if (s->get_solver_enum() == BZLA || s->get_solver_enum() == BTOR)
+  {
+    // Bitwuzla rewrite Ors as Not And
+    // it's equivalent, but cnf will be converted into a non-cnf form and this
+    // function uses the specific structure of cnf
+    return;
+  }
+  boolsort = s->make_sort(BOOL);
+  Term a = s->make_symbol("a", boolsort);
+  Term b = s->make_symbol("b", boolsort);
+  Term c = s->make_symbol("c", boolsort);
+  Term d = s->make_symbol("d", boolsort);
+  Term clause1 = s->make_term(Or, a, s->make_term(Or, b, s->make_term(Not, c)));
+  Term clause2 = s->make_term(Or, b, s->make_term(Or, s->make_term(Not, c), d));
+  Term clause3 = s->make_term(Or, d, s->make_term(Or, s->make_term(Not, c), a));
+  Term cnf=s->make_term(And, clause1, s->make_term(And, clause2, clause3));
+//The terms in the output string is not in accordance with the order of the input because of how to function is operating on the terms
```

Comment:
Is there a way to control it? Sounds like minor changes in the function will cause this test to fail. What exactly is the problem?

Reply:
(let ((.def_42 (not c))) (and (or a (or b .def_42)) (and (or b (or d .def_42)) (or d (or a .def_42)))))
Is what MSAT translated the cnf to which changed the mapping.
I tried outputting it's final string and I noticed that 'c' is being added back to the start of every clause and so it is being mapped to 1. I am thinking of only letting CVC4 test this function as it doesn't fidget with the formula and as I am using string assertion. Later when we use this formula for some other purpose we might assert something else so it won't be a problem.

Reply:
oh yes. So, maybe the best thing is to filter to only use "logging solvers" at the bottom of the test. This will ensure no rewriting or re-ordering is done.

Reply:
Is this still relevant?